src/Microsoft.AspNet.Diagnostics/StatusCodePagesExtensions.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Diagnostics;
+using Microsoft.AspNet.Http;
+
+namespace Microsoft.AspNet.Builder
+{
+    public static class StatusCodePagesExtensions
+    {
+        /// <summary>
+        /// Adds a StatusCodePages middleware with the given options that checks for responses with status codes 
+        /// between 400 and 599 that do not have a body.
+        /// </summary>
+        /// <param name="app"></param>
+        /// <param name="options"></param>
+        /// <returns></returns>
+        public static IApplicationBuilder UseStatusCodePages([NotNull]this IApplicationBuilder app, StatusCodePagesOptions options)
+        {
+            return app.UseMiddleware<StatusCodePagesMiddleware>(options);
+        }
+
+        /// <summary>
+        /// Adds a StatusCodePages middleware with a default response handler that checks for responses with status codes 
+        /// between 400 and 599 that do not have a body.
+        /// </summary>
+        /// <param name="app"></param>
+        /// <returns></returns>
+        public static IApplicationBuilder UseStatusCodePages(this IApplicationBuilder app)
+        {
+            return UseStatusCodePages(app, new StatusCodePagesOptions());
+        }
+
+        /// <summary>
+        /// Adds a StatusCodePages middleware with the specified handler that checks for responses with status codes 
+        /// between 400 and 599 that do not have a body.
+        /// </summary>
+        /// <param name="app"></param>
+        /// <param name="handler"></param>
+        /// <returns></returns>
+        public static IApplicationBuilder UseStatusCodePages(this IApplicationBuilder app, Func<StatusCodeContext, Task> handler)
+        {
+            return UseStatusCodePages(app, new StatusCodePagesOptions() { HandleAsync = handler });
+        }
+
+        /// <summary>
+        /// Adds a StatusCodePages middleware with the specified response body to send. This may include a '{0}' placeholder for the status code.
+        /// The middleware checks for responses with status codes between 400 and 599 that do not have a body.
+        /// </summary>
+        /// <param name="app"></param>
+        /// <param name="contentType"></param>
+        /// <param name="bodyFormat"></param>
+        /// <returns></returns>
+        public static IApplicationBuilder UseStatusCodePages(this IApplicationBuilder app, string contentType, string bodyFormat)
```

Comment:
Many of these overloads have similar parameters so it can be hard to pick the right one just by signature.  How about using a pattern like 'UseStatusCodePagesWithRedirects' to disambiguate?